apps/cowswap-frontend/src/modules/orders/containers/PresignedOrderNotification/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { useCallback } from 'react'
+
+import { OnPresignedOrderPayload } from '@cowprotocol/events'
+import { TokenInfo } from '@cowprotocol/types'
+
+import { EnhancedTransactionLink } from 'legacy/components/EnhancedTransactionLink'
+import { HashType } from 'legacy/state/enhancedTransactions/reducer'
+import { useOrder } from 'legacy/state/orders/hooks'
+
+import { OrderLinkWrapper } from '../../pure/commonStyled'
+import { OrderSummary } from '../../pure/OrderSummary'
+
+export interface PresignedOrderNotificationProps {
+  payload: OnPresignedOrderPayload
+  onToastMessage(message: string): void
+}
+
+export function PresignedOrderNotification(props: PresignedOrderNotificationProps) {
+  const {
+    payload: { chainId, orderUid },
+    onToastMessage,
+  } = props
+
+  const ref = useCallback(
+    (node: HTMLDivElement) => {
+      if (node) onToastMessage(node.innerText)
+    },
+    [onToastMessage]
+  )
```

Comment:
same pattern, lets make a hook :)

Reply:
Refactored in https://github.com/cowprotocol/cowswap/pull/3934